draft-ietf-quic-http.md

```diff
@@ -1926,29 +1926,55 @@ on each stream only.  As a result, if a frame type makes assumptions that frames
 from different streams will still be received in the order sent, HTTP/3 will
 break them.
 
-For example, implicit in the HTTP/2 prioritization scheme is the notion of
-in-order delivery of priority changes (i.e., dependency tree mutations): since
+Some examples of feature adaptations are described below, as well as general
+guidance to extension frame implementors converting an HTTP/2 extension to
+HTTP/3.
+
+### Prioritization Differences
+
+HTTP/2 specifies priority assignments in PRIORITY frames and (optionally) in
+HEADERS frames. Implicit in the HTTP/2 prioritization scheme is the notion of
+in-order delivery of priority changes (i.e., dependency tree mutations). Since
 operations on the dependency tree such as reparenting a subtree are not
 commutative, both sender and receiver must apply them in the same order to
 ensure that both sides have a consistent view of the stream dependency tree.
-HTTP/2 specifies priority assignments in PRIORITY frames and (optionally) in
-HEADERS frames. To achieve in-order delivery of priority changes in HTTP/3,
-PRIORITY frames are sent as the first frame on a request stream or on the
-control stream and exclusive prioritization has been removed. HTTP/3 permits the
-prioritization of requests, pushes and placeholders that each exist in separate
-identifier spaces. The HTTP/3 PRIORITY frame replaces the stream dependency
-field with fields that can identify the element of interest and its dependency.
-
-Likewise, HPACK was designed with the assumption of in-order delivery. A
-sequence of encoded header blocks must arrive (and be decoded) at an endpoint in
-the same order in which they were encoded. This ensures that the dynamic state
-at the two endpoints remains in sync.  As a result, HTTP/3 uses a modified
-version of HPACK, described in [QPACK].
+
+To achieve in-order delivery of priority changes in HTTP/3, PRIORITY frames are
+sent as the first frame on a request stream or on the control stream.  Because
+ordering between these streams is not guaranteed, the assumption that the
+PRIORITY frame on the request stream, if any, always comes first is explicitly
+stated.
+
+Exclusive prioritization has been removed, since implicitly reprioritizing other
+streams which might or might not exist yet is prone to race conditions even with
+the assumptions used to permit initial PRIORITY frames on request streams.
+
+HTTP/3 permits the prioritization of requests, pushes and placeholders that each
+exist in separate identifier spaces. The HTTP/3 PRIORITY frame replaces the
+stream dependency field with fields that can identify the element of interest
+and its dependency.
+
+### Header Compression Differences
+
+HPACK was designed with the assumption of in-order delivery. A sequence of
+encoded header blocks must arrive (and be decoded) at an endpoint in the same
```

Comment:
```suggestion
encoded header block fragments must arrive (and be decoded) at an endpoint in the same
```

Reply:
The concept of header block fragments is an h2 thing.  h3 provides the entire header block in one go, more or less, right?

Reply:
yeah I was looking at this from the wrong perspective. Header blocks is fine, I take back the suggestion.